src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt

```diff
@@ -0,0 +1,148 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.ndmis.performance
+
+import mu.KLogging
+import org.hibernate.SessionFactory
+import org.springframework.batch.core.Job
+import org.springframework.batch.core.Step
+import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory
+import org.springframework.batch.core.configuration.annotation.JobScope
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory
+import org.springframework.batch.core.job.DefaultJobParametersValidator
+import org.springframework.batch.item.ItemWriter
+import org.springframework.batch.item.database.HibernateCursorItemReader
+import org.springframework.batch.item.database.builder.HibernateCursorItemReaderBuilder
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.BatchUtils
+
+@Configuration
+@EnableBatchProcessing
+class NdmisPerformanceReportJobConfiguration(
+  private val jobBuilderFactory: JobBuilderFactory,
+  private val stepBuilderFactory: StepBuilderFactory,
+  private val batchUtils: BatchUtils,
+//  private val listener: NdmisPerformanceReportJobListener,
+  @Value("\${spring.batch.jobs.service-provider.performance-report.page-size}") private val pageSize: Int,
+  @Value("\${spring.batch.jobs.service-provider.performance-report.chunk-size}") private val chunkSize: Int,
+) {
+  @Bean
+  @JobScope
+  fun ndmisReader(
+    sessionFactory: SessionFactory,
+  ): HibernateCursorItemReader<Referral> {
+    // this reader returns referral entities which need processing for the report.
+    return HibernateCursorItemReaderBuilder<Referral>()
+      .name("ndmisPerformanceReportReader")
+      .sessionFactory(sessionFactory)
+      .queryString("select r from Referral r where sent_at is not null")
+      .build()
+  }
+
```

Comment:
writers will be updated in a following pr. This pr only focuses on the processing and logging of the rows.

Reply:
let's remove them from this PR then

future reviews are harder when the diff is commented code -> uncommented code with more changes